### PR TITLE
Migrate to ES6

### PR DESCRIPTION
**Pull request by akshatnitd**
See #613

### CODE REVIEW

**Comment by alanmoo** [comment]
Thanks. Tag me in the PRs themselves so I can add them to my review queue in the future.

**Comment by akshatnitd** [comment]
@alanmoo Sure :)

**Comment by akshatnitd** [comment]
@alanmoo @gvn Made changes in this PR as well as per the conversation in #623

File: app/react/components/tab-switcher/tab-switcher.jsx [hunk 1]
```diff
@@ -4,20 +4,34 @@ import { Link } from "react-router";
...
 
-export default React.createClass({
-  propTypes: {
+export default class TabSwitcher extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeTab: this.getSlugIndex(this.props.initialTab)
```
Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Are the `this` references why this is in a constructor?

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Same question, also should `getSlugIndex` just be passed `props` from the constructor arguments?

Comment (akshatnitd) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Yes, the better approach would be to update the `activeTab ` state in the componentWillMount lifecycle method by calling the getSlugIndex method.

@alnmoo @gvn What do you suggest?

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: That sounds like something for a follow up PR, this works for now.

Comment (akshatnitd) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Yes, will send a PR for that. :+1:

Comment (akshatnitd) [inline app/react/components/tab-switcher/tab-switcher.jsx]: @gvn @alanmoo Sent a PR #629  for this!

**Pull request merged**